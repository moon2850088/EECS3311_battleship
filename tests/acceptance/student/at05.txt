--medium sequence win
debug_test(medium)
fire([C,2])
fire([D,2])
bomb([D,2],[D,3])
bomb([A,4],[A,3])
bomb([E,4],[E,3])
fire([C,6])
fire([C,6])
fire([C,6])
debug_test(medium)
debug_test(medium)
bomb([A,2],[B,2])
bomb([A,2],[B,2])
