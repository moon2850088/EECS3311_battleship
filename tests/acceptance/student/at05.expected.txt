  state 0 OK -> Start a new game
->debug_test(medium)
  state 1 OK -> Fire Away!
     1  2  3  4  5  6
  A  _  _  _  _  _  _
  B  _  _  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([C, 2])
  state 2 OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  _  _  _  _  _  _
  B  _  _  _  _  _  _
  C  _  X  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 1/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
->fire([D, 2])
  state 3 OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  _  _  _  _  _  _
  B  _  _  _  _  _  _
  C  _  X  h  _  _  _
  D  _  X  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 2/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
->bomb([D, 2],[D, 3])
  state 4 Already fired there -> Keep Firing!
     1  2  3  4  5  6
  A  _  _  _  _  _  _
  B  _  _  _  _  _  _
  C  _  X  h  _  _  _
  D  _  X  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 2/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
->bomb([A, 4],[A, 3])
  state 5 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  _  _  O  O  _  _
  B  _  _  _  _  _  _
  C  _  X  h  _  _  _
  D  _  X  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 2/16
  Bombs: 1/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
->bomb([E, 4],[E, 3])
  state 6 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  _  _  O  O  _  _
  B  _  _  _  _  _  _
  C  _  X  h  _  _  _
  D  _  X  _  _  _  _
  E  _  v  O  O  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 2/16
  Bombs: 2/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
->fire([C, 6])
  state 7 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  _  _  O  O  _  _
  B  _  _  _  _  _  _
  C  _  X  h  _  _  O
  D  _  X  _  _  _  _
  E  _  v  O  O  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 3/16
  Bombs: 2/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
->fire([C, 6])
  state 8 Already fired there -> Keep Firing!
     1  2  3  4  5  6
  A  _  _  O  O  _  _
  B  _  _  _  _  _  _
  C  _  X  h  _  _  O
  D  _  X  _  _  _  _
  E  _  v  O  O  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 3/16
  Bombs: 2/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
->fire([C, 6])
  state 9 Already fired there -> Keep Firing!
     1  2  3  4  5  6
  A  _  _  O  O  _  _
  B  _  _  _  _  _  _
  C  _  X  h  _  _  O
  D  _  X  _  _  _  _
  E  _  v  O  O  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 3/16
  Bombs: 2/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
->debug_test(medium)
  state 10 Game already started -> Keep Firing!
     1  2  3  4  5  6
  A  _  _  O  O  _  _
  B  _  _  _  _  _  _
  C  _  X  h  _  _  O
  D  _  X  _  _  _  _
  E  _  v  O  O  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 3/16
  Bombs: 2/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
->debug_test(medium)
  state 11 Game already started -> Keep Firing!
     1  2  3  4  5  6
  A  _  _  O  O  _  _
  B  _  _  _  _  _  _
  C  _  X  h  _  _  O
  D  _  X  _  _  _  _
  E  _  v  O  O  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 3/16
  Bombs: 2/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
->bomb([A, 2],[B, 2])
  state 12 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  _  O  O  O  _  _
  B  _  O  _  _  _  _
  C  _  X  h  _  _  O
  D  _  X  _  _  _  _
  E  _  v  O  O  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 3/16
  Bombs: 3/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
->bomb([A, 2],[B, 2])
  state 13 No bombs remaining -> Keep Firing!
     1  2  3  4  5  6
  A  _  O  O  O  _  _
  B  _  O  _  _  _  _
  C  _  X  h  _  _  O
  D  _  X  _  _  _  _
  E  _  v  O  O  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 3/16
  Bombs: 3/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
